docs/open_source/setting_up/index.md

```diff
@@ -10,193 +10,131 @@ This guide focuses primarily on configuring and using various LLM clients suppor
+
 ```python
 import os
 import giskard
-import google.generativeai as genai
-from giskard.llm.client.gemini import GeminiClient
-
-# Set the Gemini API key
-os.environ["GEMINI_API_KEY"] = "…"
-
-# Configure the Gemini API
-genai.configure(api_key=os.environ["GEMINI_API_KEY"])
 
-# Create a giskard Gemini client
-gemini_client = GeminiClient()
+os.environ["GEMINI_API_KEY"] = "your-api-key"
 
-# Set the default client
-giskard.llm.set_default_client(gemini_client)
-
-# You may also want to set the default embedding model
-# Check the Custom Client Setup section for more details
+giskard.llm.set_llm_model("gemini/gemini-pro")
+# TODO: embedding
 ```
 
 ## Custom Client Setup
 
+More information on [litellm documentation](https://docs.litellm.ai/docs/providers/custom_llm_server    )
+
 ```python
 import giskard
-from typing import Sequence, Optional
-from giskard.llm.client import set_default_client
-from giskard.llm.client.base import LLMClient, ChatMessage
-
-# Create a custom client by extending the LLMClient class
-class MyLLMClient(LLMClient):
-    def __init__(self, my_client):
-        self._client = my_client
-
-    def complete(
-            self,
-            messages: Sequence[ChatMessage],
-            temperature: float = 1,
-            max_tokens: Optional[int] = None,
-            caller_id: Optional[str] = None,
-            seed: Optional[int] = None,
-            format=None,
-    ) -> ChatMessage:
-        # Create the prompt
-        prompt = ""
-        for msg in messages:
-            if msg.role.lower() == "assistant":
-                prefix = "\n\nAssistant: "
-            else:
-                prefix = "\n\nHuman: "
-
-            prompt += prefix + msg.content
-
-        prompt += "\n\nAssistant: "
-
-        # Create the body
-        params = {
-            "prompt": prompt,
-            "max_tokens_to_sample": max_tokens or 1000,
-            "temperature": temperature,
-            "top_p": 0.9,
-        }
-        body = json.dumps(params)
-
-        response = self._client.invoke_model(
-            body=body,
-            modelId=self._model_id,
-            accept="application/json",
-            contentType="application/json",
+import litellm
+from litellm import CustomLLM, completion, get_llm_provider
+
+
+class MyCustomLLM(CustomLLM):
+    def completion(self, *args, **kwargs) -> litellm.ModelResponse:
+        return litellm.completion(
+            model="gpt-3.5-turbo",
+            messages=[{"role": "user", "content": "Hello world"}],
+            mock_response="Hi!",
         )
-        data = json.loads(response.get("body").read())
 
-        return ChatMessage(role="assistant", message=data["completion"])
+    def embedding(self, *args, **kwargs) -> litellm.EmbeddingResponse:
+        return litellm.embedding(
+            model="openai/text-embedding-ada-002",
+            input=["Hello world"]
+            
+        )
```

Comment:
Here we could display a real example using `requests` lib to an external API instead of using gpt. It would be easier for the user to understand how to use it. 

In the same way, we could also create a `FakeLLM` that answers always the same content

Reply:
Yes that's true, I'll completely rewrite the LLM setup section since now we can more or less just redirect to the litellm documentation